react-projects/src/components/Calculator.js

```diff
@@ -0,0 +1,133 @@
+      setCurrentNumber('');
+      setOperation(null);
+    }
+  };
+
+  return (
+    <div className="w-full h-[900px] flex flex-col items-center justify-center">
+      <div className="w-1/4 p-7 border bg-gray-400 text-white font-bold flex justify-end">
+        {displayedNumbers || '0'}
+      </div>
+      <div className="grid grid-cols-4">
+        <button onClick={handleClick} className="p-9 border border-black font-bold" type="button">
+          AC
+        </button>
+        <button onClick={handleClick} className="p-9 border border-black font-bold" type="button">
+          +/-
+        </button>
+        <button onClick={handleClick} className="p-9 border border-black font-bold" type="button">
+          %
+        </button>
+        <button onClick={handleClick} className="p-9 bg-orange-600 text-white font-bold border border-black " type="button">
+          ÷
+        </button>
+        <button onClick={handleClick} className="p-9 border border-black font-bold" type="button">
+          7
+        </button>
+        <button onClick={handleClick} className="p-9 border border-black font-bold" type="button">
+          8
+        </button>
+        <button onClick={handleClick} className="p-9 border border-black font-bold" type="button">
+          9
+        </button>
+        <button onClick={handleClick} className="p-9 bg-orange-600 text-white font-bold border border-black" type="button">
+          x
+        </button>
+        <button onClick={handleClick} className="p-9 border border-black font-bold" type="button">
+          4
+        </button>
+        <button onClick={handleClick} className="p-9 border border-black font-bold" type="button">
+          5
+        </button>
+        <button onClick={handleClick} className="p-9 border border-black font-bold" type="button">
+          6
+        </button>
+        <button onClick={handleClick} className="p-9 bg-orange-600 text-white border font-bold border-black" type="button">
+          -
+        </button>
+        <button onClick={handleClick} className="p-9 border border-black font-bold" type="button">
+          1
+        </button>
+        <button onClick={handleClick} className="p-9 border border-black font-bold" type="button">
+          2
+        </button>
+        <button onClick={handleClick} className="p-9 border border-black font-bold" type="button">
+          3
+        </button>
+        <button onClick={handleClick} className="p-9 bg-orange-600 text-white font-bold border border-black" type="button">
+          +
+        </button>
+        <button onClick={handleClick} className="p-9 border col-span-2 border-black font-bold" type="button">
+          0
+        </button>
+        <button onClick={handleClick} className="p-9 border border-black font-bold" type="button">
+          .
+        </button>
+        <button onClick={handleClick} className="p-9 bg-orange-600 text-white border font-bold border-black" type="button">
+          =
+        </button>
```

Comment:
Remove this duplication you can use a single-button component or maybe find numbers and signs components